### PR TITLE
Improve download page

### PR DESCRIPTION
**Pull request by Holzhaus**
TODO:
- [x] Fix CSS on mobile
- [x] Add icons
- [x] ...

### CODE REVIEW

**Comment by Be-ing** [comment]
There is a lot of noise unrelated to the download page in this branch.

**Comment by Holzhaus** [comment]
> There is a lot of noise unrelated to the download page in this branch.

Yes, it's actually based on another branch that I didn't open a PR for yet.

**Comment by Holzhaus** [comment]
Okay, this is almost done. I'll resolve the merge conflict when #181 has been merged, which is a prerequisite for this PR.

File: theme/templates/pages/download.html [hunk 1]
```diff
@@ -80,9 +28,9 @@ <h1><a class="headline-link" href="#download">Download Mixxx</a></h1>
   </p>
 
   <p class="grid-container full">
-    {% with stable_version=page.stable_version %}
-    <a href="#stable" class="button button-primary download-button" title="{% trans %}Download stable version of Mixxx{% endtrans %}">{% trans %}Download {{ stable_version }}{% endtrans %}</a>
-    <span class="download-button-description">{% trans %}Select your Operating System below.{% endtrans %}</span>
+    {% with stable_version=page.versions.stable.name, stable_link="#stable" %}
+    <a href="#stable" class="button button-primary download-button" title="{% trans %}Download stable version of Mixxx{% endtrans %}" data-osdetect-text="{% trans %}Download Mixxx {{ stable_version }} for %s{% endtrans %}">{% trans %}Download Mixxx {{ stable_version }}{% endtrans %}</a>
+    <span class="download-button-description" data-osdetect-text="{% trans %}Want to download Mixxx for a different OS? Then select it manually &lt;a href=&quot;{{ stable_link }}&quot;&gt;below&lt;/a&gt;.{% endtrans %}">{% trans %}Select your Operating System <a href="{{ stable_link }}">below</a>.{% endtrans %}</span>
```
Comment (Be-ing) [inline theme/templates/pages/download.html:31]: Remove this line. It adds nothing.

File: content/pages/download.md [hunk 2]
```diff
@@ -1,39 +1,171 @@
...
+    next_stable_version: 2.3.0
+    release_announcement: /news/2020-06-07-mixxx-2-3-beta-released/
+    downloads:
+      - slug: windows
+        os: Windows 7 or later
+        packages:
+          - slug: windows32
+            name: 32-Bit
+            url: https://downloads.mixxx.org/builds/2.3/release/mixxx-2.3.0-beta-2.3-release-x86-latest.exe
+            analytics_conversion: /downloads/2.3.0-beta-win32
```
Comment (Be-ing) [inline content/pages/download.md]: remove

File: content/pages/download.md [hunk 3]
```diff
@@ -1,39 +1,171 @@
...
+        os: Ubuntu 18.04 "Bionic Beaver" or later
+        text: |
+          We provide a [PPA on Launchpad](https://launchpad.net/~mixxx/+archive/mixxxbetas) to make installing install the latest beta version of Mixxx as easy as possible.
+          Open a terminal, and enter:
+
+              $ sudo add-apt-repository ppa:mixxx/mixxxbetas
+              $ sudo apt update
+              $ sudo apt install mixxx
+
+
```
Comment (Be-ing) [inline content/pages/download.md]: Extra newline?

**Comment by Be-ing** [comment]
Thanks. This is much nicer than the walls of text we have now.

File: theme/templates/pages/download.html (hunk 1 above)
Comment (Holzhaus) [reply theme/templates/pages/download.html:31]: I think it is useful, at least when visiting the download page on Windows/macOS:

Screenshot from 2020-12-30 23-26-46

**Comment by ronso0** [comment]
yeah, only show relevant info = good design

**Comment by Holzhaus** [comment]
I also improved javascript OS detection. It now also detects Ubuntu, Arch Linux and Fedora if it's part of the browsers user-agent and links to the section.

Can someone test if this works?

1. Verify that your OS is listed in the User-Agent string: https://www.whatsmyua.info/
2. Visit the download page
    - The main download button at the top should name your OS.
    - If you click it, it should scroll to the appropriate section. On Windows/macOS it will directly download the appropriate file (on windows it will pick either 32 or 64 bit depending on your UA).

File: content/pages/download.md (hunk 2 above)
Comment (Holzhaus) [reply content/pages/download.md]: I'm in favor of removing Google Analytics, but that's  unrelated and out of scope for this PR.

**Comment by Be-ing** [comment]
The big download button at the top scrolls to here:
image
I expected it to scroll to here:
image

**Comment by Be-ing** [comment]
Clicking the headers for other OSes is now broken. The only way I can get one to expand is clicking the download button at the top.

**Comment by Holzhaus** [comment]
> Clicking the headers for other OSes is now broken. The only way I can get one to expand is clicking the download button at the top.

Oops, I messed with the label `for` attribute and broke it. Fixed now.

**Comment by Holzhaus** [comment]
> The big download button at the top scrolls to here:
> image
> I expected it to scroll to here:
> image

Actually it scrolls to the correct location, but it's behind the navbar. I now compensate for that using `scroll-margin-top`. Please test.

**Comment by Be-ing** [comment]
Now it doesn't detect my OS.

**Comment by Be-ing** [comment]
This is better, but it is still cutting off a little under the header:
image

File: content/pages/download.md [hunk 4]
```diff
@@ -1,26 +1,157 @@
...
+    name: 2.3.0 Beta
+    next_stable_version: 2.3.0
+    release_announcement: /news/2020-06-07-mixxx-2-3-beta-released/
+    downloads:
+      - slug: windows
+        os: Windows 7 or later
+        packages:
+          - slug: windows32
+            name: 32-Bit
+            url: https://downloads.mixxx.org/builds/2.3/release/mixxx-2.3.0-beta-2.3-release-x86-latest.exe
```
Comment (Be-ing) [inline content/pages/download.md]: remove

**Pull request merged**